game-app/game-headed/src/main/java/games/strategy/triplea/ui/PurchasePanel.java

```diff
@@ -53,6 +54,12 @@ public class PurchasePanel extends ActionPanel {
         public void actionPerformed(final ActionEvent e) {
           final GamePlayer player = getCurrentPlayer();
           final GameData data = getData();
+          final PurchaseDelegate purchaseDelegate = data.getPurchaseDelegate();
+
+          // load pending production from file it there is any
+          if (purchaseDelegate.getPendingProductionRules() != null) {
```

Comment:
Do we need the null check?

Just thinking this through:
  1. Initially it's already null in this class, so null check is not needed (since we'd just be assigning null to something already null).
  2. If it's not null, that means actionPerformed() already ran, but if it did, then purchaseDelegate.setPendingProductionRules() would have been called, so it should never be null.